superbench/benchmarks/micro_benchmarks/disk_performance.py

```diff
@@ -0,0 +1,215 @@
+                    ' --name=%s --rw=%s --bs=%d --time_based=1' % (io_str, fio_rw, fio_bs)
+
+    def add_parser_arguments(self):
+        """Add the specified arguments."""
+        super().add_parser_arguments()
+
+        self._parser.add_argument(
+            '--block_devices',
+            type=str,
+            nargs='*',
+            default=[],
+            required=False,
+            help='Disk block device(s) to be tested.',
+        )
+
+        # Disable precondition by default
+        self._parser.add_argument(
+            '--enable_seq_precond',
+            type=int,
+            default=0,
+            required=False,
+            help='Enable seq write precondition.',
+        )
+        self._parser.add_argument(
+            '--rand_precond_time',
+            type=int,
+            default=0,
+            required=False,
+            help='Time in seconds to run rand write precondition. Set to 0 to disable this test.',
+        )
+
+        for io_pattern in self._io_patterns:
+            for io_type in self._io_types:
+                io_str = '%s_%s' % (io_pattern, io_type)
+                self._parser.add_argument(
+                    '--%s_ramp_time' % io_str,
+                    type=int,
+                    default=self._default_ramp_time,
+                    required=False,
+                    help='Time in seconds to warm up %s test.' % io_str,
+                )
+                # Disable write tests by default
+                default_runtime = 0 if io_type == 'write' else self._default_runtime
+                self._parser.add_argument(
+                    '--%s_runtime' % io_str,
+                    type=int,
+                    default=default_runtime,
+                    required=False,
+                    help='Time in seconds to run %s test. Set to 0 to disable this test.' % io_str,
+                )
+                self._parser.add_argument(
+                    '--%s_iodepth' % io_str,
+                    type=int,
+                    default=self._default_iodepth,
+                    required=False,
+                    help='Queue depth for each thread in %s test.' % io_str,
+                )
+                default_numjobs = 1 if io_pattern == 'seq' else self._default_numjobs_for_rand
+                self._parser.add_argument(
+                    '--%s_numjobs' % io_str,
+                    type=int,
+                    default=default_numjobs,
+                    required=False,
+                    help='Number of threads in %s test.' % io_str,
+                )
+
+    def _preprocess(self):
+        """Preprocess/preparation operations before the benchmarking.
+
+        Return:
+            True if _preprocess() succeed.
+        """
+        if not super()._preprocess():
+            return False
+
+        fio_path = os.path.join(self._args.bin_dir, self._bin_name)
+
+        for block_device in self._args.block_devices:
```

Comment:
still has the issue, what if a regular file like `/etc/passwd` is provided in config, will fio throw error?
do you need to check `Path(block_device).is_block_device()`?

Reply:
This API looks great, fixed.